modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/serviceextensions/preupdatepassword/PreUpdatePasswordActionFailureTestCase.java

```diff
@@ -328,7 +338,102 @@ public void testUserSetPasswordViaOfflineInviteLink() throws Exception {
         scim2RestClient.deleteUser(offlineInvitingUserId);
     }
 
-    private void assertActionRequestPayload(String userId, String updatedPassword,
+    @Test(dependsOnMethods = "testUserSetPasswordViaOfflineInviteLink",
+            description = "Verify the admin initiated user registration with pre update password action failure")
+    public void testAdminInitiatedUserRegistration() throws Exception {
+
+        UserObject adminRegisteredUserInfo = new UserObject()
+                .userName(TEST_USER2_USERNAME)
+                .password(TEST_USER_PASSWORD)
+                .name(new Name().givenName(TEST_USER_GIVEN_NAME).familyName(TEST_USER_LASTNAME))
+                .addEmail(new Email().value(TEST_USER_EMAIL));
+        org.json.simple.JSONObject response = scim2RestClient.createUser(adminRegisteredUserInfo, null);
+
+        assertNotNull(response);
+        int statusCode = Integer.parseInt(response.get("statusCode").toString());
+        assertEquals(statusCode, expectedPasswordUpdateResponse.getStatusCode());
+
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode body = mapper.readTree(response.get("body").toString());
+
+        if (body.has("scimType")) {
+            assertEquals(body.get("scimType").asText(), String.valueOf(expectedPasswordUpdateResponse.getErrorMessage()));
+        }
+        if (statusCode == HttpServletResponse.SC_OK && body.has("detail")) {
+            assertTrue(body.get("detail").asText().contains(expectedPasswordUpdateResponse.getErrorDetail()));
+        }
+
+        assertActionRequestPayload(null, TEST_USER_PASSWORD, PreUpdatePasswordEvent.FlowInitiatorType.ADMIN,
+                PreUpdatePasswordEvent.Action.REGISTER);
+        if (statusCode == HttpServletResponse.SC_CREATED && body.has("id")) {
+            scim2RestClient.deleteUser(body.get("id").asText());
+        }
```

Comment:
This is the failure test case right ? 
Then why do we need to delete the user as the user is not created.